### ISSUE TITLE
Created files are invalid

### ISSUE THREAD

**Issue by cloudlena** [tormoder/fit#69]
## Problem on Strava

The files created by `Encode` are somehow invalid for some platforms. For example, if I upload the newly created file to Strava, it says "The upload contains bad data.".

I tried to debug them by having a valid FIT file (e.g., https://github.com/tormoder/fit/blob/master/testdata/me/activity-small-fenix2-run.fit) and running it through the following simple program:

```go
package main

import (
	"bytes"
	"encoding/binary"
	"log"
	"os"

	"github.com/tormoder/fit"
)

func main() {
	testData, err := os.ReadFile("activity-small-fenix2-run.fit")
	if err != nil {
		log.Fatalln(err)
	}

	file, err := fit.Decode(bytes.NewReader(testData))
	if err != nil {
		log.Fatalln(err)
	}

	f, err := os.Create("new.fit")
	if err != nil {
		log.Fatalln(err)
	}
	defer f.Close()

	err = fit.Encode(f, file, binary.LittleEndian)
	if err != nil {
		log.Fatalln(err)
	}
}

```

## Using FitFileViewer

When uploading the `new.fit` file to https://www.fitfileviewer.com/, it shows empty data (whereas the original `activity-small-fenix2-run.fit` file showed valid content):

![screenshot-2022-09-04-19-49-09](https://user-images.githubusercontent.com/6780471/188326773-75bce5c5-8d21-4032-8e03-61ebb483b8ad.png)

## Using GPXSee

Weirdly enough, https://www.gpxsee.org/ shows the `new.fit` file correctly.

## Using RUNALYZE

Analyzing both files with https://runalyze.com/tool/fit-viewer, the main differences seem to be that `new.fit` has a lot of `enhanced_speed`, `enhanced_max_speed`, `enhanced_altitude` fields on its laps and records (whereas `activity-small-fenix2-run.fit` doesn't have any `enhanced_*` fields) and that `new.fit` always uses `0` as its local message type index.

Update: I did another test run where I removed the code that generates the `enhanced_*` field generation code from this library and even without those, neither Strava nor FitFileViewer would accept the file. So I'm guessing that those aren't the problem.

**Comment by cloudlena** [comment]
For reference, here are both FIT files:

[files.zip](https://github.com/tormoder/fit/files/9487456/files.zip)

**Comment by cloudlena** [comment]
I just did another test, uploading the newly created file to https://www.fitfiletools.com/#/remover#remover. There, the shown seems to be very weird (see screenshot):

![screenshot-2022-09-05-09-41-37](https://user-images.githubusercontent.com/6780471/188395554-2b820c0b-4f8b-4e75-9e77-954549d9f48c.png)

I'm thinking, the issue might be that the order of the created records is not kept.

**Comment by cloudlena** [comment]
Uploading the `new.fit` file to http://garmin.kiesewetter.nl/ yields the following error: `Upload status: The file could not be uploaded. The following error occured: Unexpected end of file while parsing Name has occurred. Line 151558, position 9.`.

**Comment by tormoder** [comment]
Thanks for the report.

I'll try to look into this if I have some available time (together with your other PRs), but it may take a while.

**Comment by cloudlena** [comment]
Thanks, @tormoder! Is there any way I can support more or any more information you might need?

**Comment by cloudlena** [comment]
@tormoder are there any news on this issue? Can I support in any way?

**Comment by tormoder** [comment]
Just released `v0.15.0` with #67 included.

@cloudlena, if you're still interested, you can check if that change solves your issue.

**Comment by cloudlena** [comment]
Thanks, @tormoder. Indeed, that fixed it for me :tada:
